pages/compare.js

```diff
@@ -1,26 +1,227 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+// Modules
+import { getSpecificIndicators, setIndicatorsLayersActive, setIndicatorsLayers } from 'modules/indicators';
+import {
+  setSingleMapParams,
+  setSingleMapParamsUrl,
+  addArea,
+  removeArea
+} from 'modules/maps';
+
+// Selectors
+import { getIndicatorsWithLayers } from 'selectors/indicators';
+
 // Redux
 import withRedux from 'next-redux-wrapper';
 import { store } from 'store';
 
+// Libraries
+import isEqual from 'lodash/isEqual';
+
+// Utils
+import { setLayersZIndex } from 'utils/map';
+import { decode } from 'utils/general';
+
 // Components
 import Page from 'components/layout/page';
 import Layout from 'components/layout/layout';
+import Accordion from 'components/ui/accordion';
+import Map from 'components/map/map';
+import Legend from 'components/map/legend';
+
+// Constants
+import { MAP_OPTIONS } from 'constants/map';
+
+
+// let L;
+// if (typeof window !== 'undefined') {
+//   /* eslint-disable global-require */
+//   L = require('leaflet/dist/leaflet');
+//   /* eslint-enable global-require */
+// }
 
 class ComparePage extends Page {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      hidden: []
+    };
+
+    this.url = props.url;
+
+    // Bindings
+    this.onToggleAccordionItem = this.onToggleAccordionItem.bind(this);
+    this.onAddArea = this.onAddArea.bind(this);
+  }
+
+  /* Lyfecycle */
+  componentDidMount() {
+    const { url } = this.props;
+
+    this.props.getSpecificIndicators(url.query.indicators);
+
+    if (url.query.maps) {
+      const params = decode(url.query.maps);
+      this.setMapParams(params);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!isEqual(this.props.url.query, nextProps.url.query)) {
+      this.url = nextProps.url;
+    }
+  }
+
+  /* Accordion methods */
+  onToggleAccordionItem(e, id) {
+    let newHidden = this.state.hidden;
```

Comment:
Is hidden an array??? You are committing a crime here. Remember that we should always create a new array